### PR TITLE
Add support for vectors, decimals, and nullables to Spark TensorFlow connector

### PR DESCRIPTION
**Pull request by skavulya**

### CODE REVIEW

**Comment by skavulya** [comment]
Thanks, Joyesh.

**Comment by skavulya** [comment]
@jhseu Joyesh and I have submitted two pull request https://github.com/tensorflow/ecosystem/pull/53 https://github.com/tensorflow/ecosystem/pull/54 to address issues https://github.com/tensorflow/ecosystem/issues/42 https://github.com/tensorflow/ecosystem/issues/43. Please review the PRs for us, and let us know if we need to make any changes. Thanks.

File: spark/spark-tensorflow-connector/src/main/scala/org/tensorflow/spark/datasources/tfrecords/serde/DefaultTfRecordRowEncoder.scala [hunk 1]
```diff
@@ -111,9 +124,20 @@ object DefaultTfRecordRowEncoder extends TfRecordRowEncoder {
...
+      case ArrayType(DecimalType(), _) =>
+        val decimalArray = ArrayData.toArrayData(row.get(index)).toArray[Decimal](DataTypes.createDecimalType())
+        FloatListFeatureEncoder.encode(decimalArray.map(_.toFloat))
       case ArrayType(_, _) =>
         BytesListFeatureEncoder.encode(ArrayData.toArrayData(row.get(index)).toArray[String](StringType))
-      case _ => BytesListFeatureEncoder.encode(Seq(row.getString(index)))
+      case VectorType => {
+        val field = row.get(index)
+        field match {
+          case v: SparseVector => FloatListFeatureEncoder.encode(v.toDense.toArray.map(_.toFloat))
```
Comment (pilipolio) [inline spark/spark-tensorflow-connector/src/main/scala/org/tensorflow/spark/datasources/tfrecords/serde/DefaultTfRecordRowEncoder.scala]: wouldn't `tf.VarLenFeature` be a better match for SparseVector? The `toDense` will be very inefficient for sparse features with large vocabulary.

Comment (skavulya) [inline spark/spark-tensorflow-connector/src/main/scala/org/tensorflow/spark/datasources/tfrecords/serde/DefaultTfRecordRowEncoder.scala]: @pilipolio The tf.VarLenFeature is only available in the TensorFlow Python API. We could use the TensorFlow java API, but it was not available when we initially created the library and would involve a major re-write which we can address in a different PR.

Comment (pilipolio) [inline spark/spark-tensorflow-connector/src/main/scala/org/tensorflow/spark/datasources/tfrecords/serde/DefaultTfRecordRowEncoder.scala]: 👍 Eager to see this functionality in the future

**Comment by skavulya** [comment]
@jhseu There are a few PRs for the Spark TensorFlow connector which address pending issues. Please review them, and let us know if we should make any changes. Thanks!

**Comment by jhseu** [comment]
Sorry for the delay on these. Been swamped the past month :) Going to just quickly skim over them.

**Comment by skavulya** [comment]
@jhseu No problem. Thanks for merging the PRs :)

**Pull request merged**